### DISCUSSION TITLE
Setup Manager v1.1beta

### DISCUSSION THREAD

**Discussion by scriptingosx** [jamf/Setup-Manager#29] [Announcements]
### New Features

- new action [`waitForUserEntry`](ConfigurationProfile.md#waitforuserentry) which allows for two-phase installation workflows in Jamf Pro. When Setup Manager reaches this action it will wait for the user entry to save the data entry, then it will run a recon/Update Inventory. Policy actions that follow this, can then be scoped to data from the user entry. (#11)
- Setup Manager can now run over Login Window, instead of immediately after installation. This also allows Setup Manager to work with AutoAdvance. Use the new `runAt` key in the profile to determine when Setup Manager runs (#18)
- data from user entry, is now written to a file when Setup Manager submits data. See details in [User Entry](Extras.md#user-data-file) (#9)
- use token substitution in the `title`, `message`, and action `label` values (as well as `computerNameTemplate`)
- token substitution can extract center characters with `:=n`
- localization in the configuration profile has been simplified. The previous method still works, but is considered deprecated. [Details](ConfigurationProfile.md#localization)

### Fixes and improvements

- icons using `symbol:` that end in `.app` now work properly
- Elapsed time is shown in "About this Mac…" Start time is shown with option key
- svg and pdf images used for `icon`s should now work
- general fixes in user entry setup
- improved rendering in Help View (#12)
- fixes to json schema
- improved and updated documentation
- included Installomator script updated to [v10.6](https://github.com/Installomator/Installomator/releases/v10.6)
- added Setup Manager version and macOS version and build to tracking ping
- fixed UI glitch in macOS Sequoia

<hr /><em>This discussion was created from the release <a href='https://github.com/Jamf-Concepts/Setup-Manager/releases/tag/v1.1beta'>Setup Manager v1.1beta</a>.</em>

**Comment by TechTrekkie** [comment]
Does running at the login window work with Jamf Connect? I've added the runAt key and set it to loginwindow but it won't launch.

**Reply by scriptingosx** [reply]
This is not something we have tested. I would recommend to not install JamfConnect before Setup Manager got a chance to run. Then Setup Manager can install and configure Jamf Connect

**Reply by TechTrekkie** [reply]
OK so that seemed to work, however, when Setup Manager gets to the step to install Jamf Connect, it kills setup manager, the screen goes back for about 20 seconds and then the Jamf Connect login screen appears.

**Reply by TechTrekkie** [reply]
actually scratch that.... setup manager isnt launching at the login window right after setup assistant, but if I reboot then it will launch when it gets back to the login screen.

**Reply by sbickfo2** [reply]
experiencing the same issue as the comment above, setup manager when set to runAt loginwindow does not start unless the machine is rebooted after setup assistant. No jamf connect being installed, just a local admin account via the prestage.

**Reply by scriptingosx** [reply]
What does `/Library/Logs/Setup Manager.log` say?

**Reply by sbickfo2** [reply]
Here is what the logs say when enrollment finishes and I just log into the machine, I can also get the setup manager logs after rebooting when setup manager actually starts if that would help

8:12:14 AM preventing display and system sleep
8:12:14 AM runAt: loginwindow, console user: _mbsetupuser
8:12:19 AM retry: runAt: loginwindow, console user: _mbsetupuser
8:12:37 AM preventing display and system sleep
8:12:37 AM runAt: loginwindow, console user: _mbsetupuser
8:12:42 AM retry: runAt: loginwindow, console user: <nil>

**Comment by bghilardibm** [comment]
Hello,
I don't understand why the pngs are not read.
I put them in the configuration profile. They are well loaded with a pkg.
<key>background</key>
        <string>/Library/BM/illustration.png</string>
        <key>message</key>
	<string>Veuillez patienter</string>
	<key>icon</key>
	<string>/Library/BM/logo.png</string>

**Reply by bghilardibm** [reply]
<img width="257" alt="Capture d’écran 2024-10-25 à 12 14 43" src="https://github.com/user-attachments/assets/cd871733-fd57-4074-a18e-e2388310b5f3">
